Makefile

```diff
@@ -56,7 +56,8 @@ prefix_install: installables
 	mkdir -p "$(PREFIX)/Frameworks" "$(PREFIX)/bin"
 	rsync -a --delete "$(TEMPORARY_FOLDER)$(FRAMEWORKS_FOLDER)/$(OUTPUT_FRAMEWORK)" "$(PREFIX)/Frameworks/"
 	cp -f "$(TEMPORARY_FOLDER)$(BINARIES_FOLDER)/carthage" "$(PREFIX)/bin/"
-	install_name_tool -rpath "/Library/Frameworks" "@executable_path/../Frameworks/$(OUTPUT_FRAMEWORK)/Versions/Current/Frameworks/"  "$(PREFIX)/bin/carthage"
+	install_name_tool -rpath "/Library/Frameworks" "@executable_path/../Frameworks" "$(PREFIX)/bin/carthage"
```

Comment:
I recommend checking for each installation method supported by carthage, whether the frameworks are loaded from the place where you expect.
```terminal.sh-session
➜  9:09:15 git:(improve-makefile-rpath-handling) ✗ lldb prefix/bin/carthage
(lldb) target create "prefix/bin/carthage"
Current executable set to 'prefix/bin/carthage' (x86_64).
(lldb) image list
[  0] BB6A53CE-A026-321A-BF05-1D15E0F694C3 0x0000000100000000 /Users/norio/github/Carthage/prefix/bin/carthage 
[  1] 322C06B7-8878-311D-888C-C8FD2CA96FF3 0x0000000000000000 /usr/lib/dyld 
[  2] 730B7944-BB43-35D5-A546-9F6CCED4B9F3 0x0000000000000000 /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation 
[  3] 54CD8D1A-5C98-3559-B13A-932B3D3DD338 0x0000000000000000 /usr/lib/libobjc.A.dylib 
[  4] FC9E9F13-3B18-305C-BE0A-97C7843652B0 0x0000000000000000 /usr/lib/libSystem.B.dylib 
[  5] 36385A3A-60D3-32DB-BF55-C6D8931A7AA6 0x0000000000000000 /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation 
[  6] FC732A82-DA1D-3FAE-98FA-DF74A3259547 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/CarthageKit 
[  7] BAADDD8C-01DA-3D2D-AB7B-44CE0ED5C71B 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/Commandant.framework/Commandant 
[  8] 58E9C9D9-F591-355D-B713-66E8D306DD7B 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/PrettyColors.framework/Versions/A/PrettyColors 
[  9] 2BFEFBE7-B50D-3982-9CAF-3645848CBF82 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/ReactiveSwift.framework/ReactiveSwift 
[ 10] DE0502E9-7A2D-346C-990B-BF8315064450 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/ReactiveTask.framework/ReactiveTask 
[ 11] CB71357E-9434-3003-AD79-575A39BA6006 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/Result.framework/Versions/A/Result 
[ 12] 7743D52D-0B42-3122-A62A-111239527FE9 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/Tentacle.framework/Versions/A/Tentacle 
[ 13] E48ADED4-42C5-3359-9055-614B4E524BE2 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/XCDBLD.framework/XCDBLD 
[ 14] F3EE8627-3CB7-3ECE-B79D-9CA8298466A9 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftAppKit.dylib 
[ 15] F3C2AEFF-073B-3D90-9F72-0CAA3B814A23 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftCore.dylib 
[ 16] 1A670783-4A8C-37E3-A907-50A677DB7D5E 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftCoreData.dylib 
[ 17] 739DD9C7-DB67-3B64-A139-48CACD44741A 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftCoreGraphics.dylib 
[ 18] 07D26DC6-DC2F-3966-BC87-0EA6FC215C07 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftCoreImage.dylib 
[ 19] A5DE2CCA-D683-31C5-A6B4-1DC5831D0EF3 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftDarwin.dylib 
[ 20] CB9C8886-19E3-352A-B8C4-02C7ECD18738 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftDispatch.dylib 
[ 21] 7FED0AEF-4FBE-3EDA-AAB3-41262EC6B33E 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftFoundation.dylib 
[ 22] 9408AC1E-676A-32B8-8E1A-4F2A4D5675BD 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftIOKit.dylib 
[ 23] C5FD2292-A97E-33FD-B1F8-02C9250021BA 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftObjectiveC.dylib 
[ 24] D9A41D53-AA11-30B5-8247-9B5BFAACD7F0 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftQuartzCore.dylib 
[ 25] 5ED57C16-AB3A-3B11-A714-C246CCCCFE82 0x0000000000000000 /Users/norio/github/Carthage/prefix/Frameworks/CarthageKit.framework/Versions/Current/Frameworks/libswiftXPC.dylib 
```